### PR TITLE
chore: Upgrade react-native-svg to 15.3

### PR DESCRIPTION
**Pull request by smilingkylan**
## **Related issues**

Fixes: https://github.com/orgs/MetaMask/projects/60/views/6?pane=issue&itemId=61547905

## **Manual testing steps**

1. Go to https://app.poolside.party with the in-app browser and connect the wallet
2. Once you connect, you would see an `Root Error: Text strings must be rendered within a <Text> component.` error message, but **now** you should not see that error message... the wallet should connect just fine.

### **Before**
https://github.com/MetaMask/metamask-mobile/assets/6249205/6a58eca9-e9a0-4240-acdd-f7fb3b52f033

### **After**
https://github.com/MetaMask/metamask-mobile/assets/6249205/d9e68e1f-6aff-42bb-9b3f-80eae8ef2e2d

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [X] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [X] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1487,7 +1487,11 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
-				OTHER_LDFLAGS = "$(inherited)";
+				OTHER_LDFLAGS = (
+					"$(inherited)",
+					"-Wl",
+					"-ld_classic",
```
Comment (legobeat) [inline ios/MetaMask.xcodeproj/project.pbxproj:1493]: Why is this added here?

Comment (kylanhurt) [reply ios/MetaMask.xcodeproj/project.pbxproj:1493]: @legobeat It was a bug created elsewhere that makes the change when we run `yarn setup`. The PR was merged into main yesterday and I have brought that fix into this branch to get rid of the change.

PR in question: https://github.com/MetaMask/metamask-mobile/pull/9657

**Comment by kylanhurt** [comment]
@SocketSecurity ignore npm/react-native-svg@15.3.0

Latest contributor appears to be part of the organization that maintains the GitHub repo and npm package. Also contributor to other React Native projects

**Review by legobeat — APPROVED** [review]
LGTM
